json-format.md

```diff
@@ -129,34 +129,85 @@ OPTIONAL not omitted attributes MAY be represeted as a `null` JSON value.
 
 ### 3.1. Handling of "data"
 
+The JSON representation of the event "data" payload is determined by the runtime
+type of the `data` content and the value of the [`datacontenttype`
+attribute][datacontenttype].
+
+#### 3.1.1. Payload Serialization
+
 Before taking action, a JSON serializer MUST first determine the runtime data
 type of the `data` content.
 
 If the implementation determines that the type of data is `Binary`, the value
 MUST be represented as a [JSON string][json-string] expression containing the
 [Base64][base64] encoded binary value, and use the member name `data_base64` to
-store it inside the JSON object.
-
-For any other type, the implementation MUST translate the data value into a
-[JSON value][json-value], and use the member name `data` to store it inside the
-JSON object.
-
-Out of this follows that use of the `data` and `data_base64` members is mutually
-exclusive in a JSON serialized CloudEvent.
+store it inside the JSON representation. If present, the `datacontenttype` MUST
+reflect the format of the original binary data.
+
+If the type of data is not `Binary`, the implementation will next determine
+whether the value of the `datacontenttype` attribute declares the `data` to
+contain JSON-formatted content. Such a content type is defined as one having a
+[media subtype][rfc2045-sec5] equal to `json` or ending with a `+json` format
+extension. That is, a `datacontenttype` declares JSON-formatted content if its
+media type, when stripped of parameters, has the form `*/json` or `*/*+json`.
+
+If the `datacontenttype` declares the data to contain JSON-formatted content, a
+JSON serializer MUST translate the data value to a [JSON value][json-value], and
+use the member name `data` to store it inside the JSON representation. The data
+value MUST be stored directly as a JSON value, rather than as an encoded JSON
+document represented as a string. An implementation MAY fail to serialize the
+event if it is unable to translate the runtime value to a JSON value.
+
+Otherwise, if the `datacontenttype` does not declare JSON-formatted data
+content, a JSON serializer MUST store a string representation of the data value,
+properly encoded according to the `datacontenttype`, in the `data` member of the
+JSON representation. An implementation MAY fail to serialize the event if it is
+unable to represent the runtime value as a properly encoded string.
+
+Out of this follows that the presence of the `data` and `data_base64` members is
+mutually exclusive in a JSON serialized CloudEvent.
+
+Furthermore, unlike attributes, for which value types are restricted to strings
```

Comment:
Values aren't restricted to strings under the type system mapping - Boolean attributes are represented as Boolean JSON values, and Integer attributes are represented as number JSON values. (This was a bug in the previous version of the document, by the looks of it.)

Possibly just:

> Furthermore, unlike attributes, the `data` member JSON value is unrestricted [...]

(I don't think going into *more* detail about which types are permitted by the mapping would be useful.)

Reply:
Good catch on that. Fixed.